packages/test-scenarios/__tests__/scenario-test.js

```diff
@@ -0,0 +1,174 @@
+const { Scenarios, Project } = require('scenario-tester');
+const { dirname, delimiter } = require('path');
+const { merge } = require('lodash');
+
+jest.setTimeout(500000);
+
+// https://github.com/volta-cli/volta/issues/702
+// We need this because we're launching node in child processes and we want
+// those children to respect volta config per project.
+(function restoreVoltaEnvironment() {
+  let voltaHome = process.env['VOLTA_HOME'];
+  if (!voltaHome) return;
+  let paths = process.env['PATH'].split(delimiter);
+  while (/\.volta/.test(paths[0])) {
+    paths.shift();
+  }
+  paths.unshift(`${voltaHome}/bin`);
+  process.env['PATH'] = paths.join(delimiter);
+})();
+
+// eslint-disable-next-line node/no-unpublished-require
+const babelPluginPath = require.resolve('../../babel-plugin-ember-test-metadata/dist/index');
+
+async function classic(project) {
+  merge(project.files, {
+    'ember-cli-build.js': `'use strict';
+
+const EmberApp = require('ember-cli/lib/broccoli/ember-app');
+
+module.exports = function (defaults) {
+  let app = new EmberApp(defaults, {
+    babel: {
+      plugins: [
+        [
+          '${babelPluginPath}',
+          { enabled: true }
+        ]
+      ],
+    }
+  });
+
+  return app.toTree();
+};
+`,
+  });
+}
+
+async function embroider(project) {
+  project.linkDependency('@embroider/core', {
+    baseDir: __dirname,
+  });
+  project.linkDependency('@embroider/compat', {
+    baseDir: __dirname,
+  });
+  project.linkDependency('@embroider/webpack', {
+    baseDir: __dirname,
+  });
+  project.linkDependency('webpack', {
+    baseDir: __dirname,
+  });
+
+  merge(project.files, {
+    'ember-cli-build.js': `'use strict';
+
+const EmberApp = require('ember-cli/lib/broccoli/ember-app');
+
+module.exports = function (defaults) {
+  let app = new EmberApp(defaults, {
+    babel: {
+      plugins: [
+        [
+          '${babelPluginPath}',
+          { enabled: true }
+        ]
+      ],
+    }
+  });
+
+  const { Webpack } = require('@embroider/webpack');
+  return require('@embroider/compat').compatBuild(app, Webpack);
+};
+`,
+  });
+}
+
+function baseApp() {
+  return Project.fromDir(
+    // eslint-disable-next-line node/no-unpublished-require
+    dirname(require.resolve('@babel-plugin-ember-test-metadata/app-template/package.json')),
+    {
+      linkDeps: true,
+    }
+  );
+}
+
+Scenarios.fromProject(baseApp)
+  .expand({
+    classic,
+    embroider,
+  })
+  .map('app scenarios', (project) => {
+    merge(project.files, {
+      tests: {
+        unit: {
+          'with-hooks-test.js': `import { module, test } from 'qunit';
+import { getTestMetadata } from '@ember/test-helpers';
+
+module('Acceptance | with-hooks-test', function (hooks) {
+  hooks.beforeEach(function () {
+    // noop
+  });
+
+  test('example', async function (assert) {
+    assert.ok(getTestMetadata(this).filePath.includes('tests/unit/with-hooks-test.js'));
```

Comment:
These assertions should be changed to an equals in order to correctly check the path is correct.

Reply:
acknowledged